src/libsyntax/error_codes.rs

```diff
@@ -363,6 +363,36 @@ and likely to change in the future.
 
 "##,
 
+E0704: r##"
+This error indicates that a incorrect visibility restriction was specified.
+
+Example of erroneous code:
+
+```compile_fail,E0704
+mod foo {
+    pub(foo) struct Bar {
+        x: i32
+    }
+}
+```
+
+To make struct `Bar` only visible in module `foo` the `in` keyword should be
+used:
+```
+fn main() {
+    mod foo {
+        pub(in crate::foo) struct Bar {
```

Comment:
To make `crate::foo` resolve, it needs to be outside `fn main`:

```rust
mod foo {
    pub(in crate::foo) struct Bar {
        x: i32,
    }
}

fn main() {}
```

Reply:
Thanks man :) I was confused because of the example. But I committed your suggestion!

Reply:
I think that the suggested code was for an empty main after all the code you wrote, not to enclose your code in main, which is what rustdoc already does behind the scenes. You can even hide the `fn main() {}` from rustdoc output by using a leading `#`.

Reply:
Ahhh sorry! So if I understand correctly I should change:

```fn main() {}```
to
```# fn main() {}```

Reply:
That would hide that line, yes. It's used to hide parts of code that are needed for the doctest to run correctly, but not for understanding the code (things like setup, imports and this case of a defined main fn).

Reply:
Ahh thanks! I changed the code now such that it is commented. I'll check again tomorrow if bors agrees with us as well and then I'll squash my commits again (Y).